aip/auth/4115.md

```diff
@@ -5,124 +5,115 @@ state: approved
-credential flow, please read [AIP-4110][2].
+1. Retrieving a configuration through a metadata server (MDS) endpoint. The configuration specifies 
+   how to access Google’s default mTLS credentials.
+2. Requesting bound tokens.
 
-### Google Virtual Machine Environments
+**Note:** Because this AIP describes guidance and requirements in a language-neutral way, it uses 
+generic terminology which may be imprecise or inappropriate in certain languages or environments.
 
-There are typically two types of Google virtual machine environments where the
-auth library should obtain the identity token based on the environment type:
+## Prerequisites
 
-- Google App Engine Standard 1.0 (aka 1st generation)
-- Compute Engine 
-  - GCE equivalent runtimes
-    - Google App Engine Standard 2.0+
-    - Google App Engine Flex
-    - Google Cloud Functions
-    - Cloud Run
-    - Workload Identity on Google Kubernetes Engine
+Before trying to use Google’s default mTLS credentials and bound tokens, the client **must** first 
+check if the remote Google API endpoint supports mTLS. If the remote endpoint does NOT support mTLS,
+the client **should** connect to the endpoint using TLS and request a token from MDS. How to check 
+if an endpoint supports mTLS is out of the scope of this AIP. If the remote endpoint does support 
+mTLS, the client **should** try to connect using mTLS first before falling back to TLS. How to find 
+the remote API’s mTLS endpoint is out of the scope of this AIP.
 
-The auth library **should** depend on the [Google App Engine SDK][3] or well 
-defined GAE environment variables to detect if the application is running within
-the 1st generation Google App Engine environment.
-
-To detect if the application is running on Compute Engine or an equivalent runtime,
-the auth library **should** depend on the [Metadata Service Library][4].
+## Guidance
 
-### Compute Engine or Equivalent Runtime
+### Access Default mTLS Credentials
 
-If the application runs on a Compute Engine instance (or equivalent runtime),
-the auth library can request the following URL to get an access token associated
-with instance:
+To leverage Google’s default mTLS credentials, the client **should** retrieve configurations from 
+MDS. The MDS in all virtual environments (GCE, Serverless, and GKE) expose an HTTP endpoint that 
+serves a configuration that specifies how to access Google's default mTLS credentials. This endpoint 
+is called the mTLS configuration endpoint.
 
+The URL of the MDS's mTLS configuration endpoint is: 
 ```
-http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token
+http://metadata.google.internal/computeMetadata/v1/googleAutoMtlsConfiguration
 ```
 
-In response, the auth library will usually receive a token in the following JSON
-format:
+The request to the MDS's mTLS configuration endpoint **should** be an HTTP GET request without any 
+parameter or payload.
 
-```json
```

Comment:
Should this information be kept in the AIP?

Reply:
Done. Good idea.